### PR TITLE
[Relax] Batch norm correctness on eval mode

### PR DESCRIPTION
**Pull request by hugolatendresse**
Batch_norm is a different operator in training and eval. The previous interface defaulted to the training mode and required changing an ingested pytorch program itself to use the eval mode. This is sub-ideal, especially since torch.export explicitely communicates whether batch_norm should be in training or eval in a given torch program. 

This PR automates the selection of training/eval mode in the exported program translator, and achieves correctness for eval mode. 

Future TODO: there is something wrong with batch_norm on training mode. It does not pass a correctness test when taken straight from the main branch (there's an issue with tensor dimensions). I added a note to address later as training mode is probably not high priority.

### CODE REVIEW

**Comment by hugolatendresse** [comment]
cc: @MasterJH5574 this is ready for review

**Comment by hugolatendresse** [comment]
@tvm-bot rerun

**Review by MasterJH5574 — APPROVED** [review]
Looks good. Thank you @hugolatendresse for the enhancement!

File: python/tvm/relax/frontend/torch/exported_program_translator.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def _hardtanh(self, node: fx.Node) -> relax.Expr:
 
     ########## Neural Network ##########
 
-    def _batch_norm_legit_no_training(self, node: fx.Node) -> relax.Var:
+    def _batch_norm(self, node: fx.Node, training) -> relax.Var:
```
[the remark this answers is not in the document]
Comment (hugolatendresse) [reply python/tvm/relax/frontend/torch/exported_program_translator.py:48]: Got it, will do, thanks
